azkaban-common/src/main/java/azkaban/executor/container/watch/KubernetesWatch.java

```diff
@@ -77,7 +79,12 @@ public KubernetesWatch(ApiClient apiClient,
     this.apiClient = apiClient;
     // no timeout for request completion
     OkHttpClient httpClient =
-        this.apiClient.getHttpClient().newBuilder().readTimeout(0, TimeUnit.SECONDS).build();
+        this.apiClient.getHttpClient().newBuilder()
+            .protocols(Arrays.asList(Protocol.HTTP_2,Protocol.HTTP_1_1))
+            .connectTimeout(60, TimeUnit.SECONDS)
+            .callTimeout(1200, TimeUnit.SECONDS)
```

Comment:
Shall we introduce these timeouts from config? Just thinking out loud.

Reply:
This is ideally not even needed since this newer client handles this for us. However if we do decide to place it , I shall include it via a config.